src/Icon/Icon.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import FontAwesome from 'react-fontawesome'
+
+const Icon = ({ name, ...props }) => {
+  if (name) {
+    return <FontAwesome name={name} {...props} />
+  } else {
+    return <span aria-hidden="true" {...props} />
+  }
+}
+Icon.propTypes = {
+  /** Font awesome icon rendered if name prop is provided */
+  name: PropTypes.string
+}
+export default Icon
```

Comment:
I am suggesting to add a type prop that can be `fa` or `pf`
It will make the API around the Icon easier to understand

```jsx
// In different file
const PatternflyIcon = ({ name, className, props }) => { 
  return (<span
    aria-hidden="true"
    className=cx('pficon', `pficon-${name}`, className)
    {...props}
  />);
}

PatternflyIcon.propTypes = {
  name: PropTypes.string.isRequired
}
  
const Icon = ({ type, ...props }) => {
  if (type === 'fa') {
    return <FontAwesome {...props} />
  }

  if (type === 'pf') {
    return <PatternflyIcon {...props} />
  }

  throw new Error(`Unsupported prop type=${type}`)
}

Icon.propTypes = {
  type: PropTypes.oneOf(['fa', 'pf']),
  name: PropTypes.string.isRequired
}

ListViewIcon.defaultProps = {
  type: 'fa'
}
```

Reply:
I like it... I've updated it just now.